### PR TITLE
[PA-2] modularized dataset, model, utils

### PR DESCRIPTION
**Pull request by mrcolo**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,204 +1,32 @@
...
-    )
-
-    # tensorboard callback
-    logdir = "logs/scalars/{}".format(dt.today())
-    tensorboard_cb = tf.keras.callbacks.TensorBoard(log_dir=logdir)
-    gb = GetWeights()
-    # get model
+    
+    # set up
+    t_ds, v_ds, window = get_datasets()
```
Comment (ElliotVilhelm) [inline main.py:19]: lets not pass window here

File: main.py [hunk 2]
```diff
@@ -1,204 +1,32 @@
...
-    gb = GetWeights()
-    # get model
+    
+    # set up
+    t_ds, v_ds, window = get_datasets() 
     ssm = get_lstm()
 
     # run trial
     history = ssm.fit(t_ds, epochs=EPOCHS,
                       steps_per_epoch=window,
```
Comment (ElliotVilhelm) [inline main.py:24]: we only need it here so lets calculate it if we can

File: dataset.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                                        STEP)
+
+    # construct datasets
+    t_ds = tf.data.Dataset.from_tensor_slices((xt, yt))
+    t_ds = t_ds.cache().shuffle(
+        BUFFER_SIZE).batch(BATCH_SIZE).repeat()
+    v_ds = tf.data.Dataset.from_tensor_slices((xv, yv))
+    v_ds = v_ds.batch(BATCH_SIZE).repeat()
+    
+    return t_ds, v_ds, window
```
Comment (ElliotVilhelm) [inline dataset.py:112]: new line pls

File: utils.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import numpy as np
+import tensorflow as tf
+def create_time_steps(length):
+    return list(range(-length, 0))
+
+class GetWeights(tf.keras.callbacks.Callback):
```
Comment (ElliotVilhelm) [inline utils.py:6]: this is a call back as well, should go in callbacks.py

Comment (mrcolo) [reply utils.py:6]: u right

File: dataset.py (hunk 3 above)
Comment (mrcolo) [reply dataset.py:112]: yessir

File: main.py (hunk 2 above)
Comment (ElliotVilhelm) [reply main.py:24]: I think maybe hard with tfds to get len of original dataset

**Pull request merged**